### PR TITLE
layers: Update 02697 logging for GPLs

### PR DESCRIPTION
**Pull request by ncesario-lunarg**
Ensure valid pipeline layout handles get printed when logging errors
involving graphics pipeline libraries containing merged layouts.

When using graphics pipeline libraries, it is possible that the "executable pipeline" used at draw time is actually a union of layouts from pre-raster and fragment shader libraries. If this is the case, then currently no information about the original layouts are printed to the user.

### CODE REVIEW

File: layers/pipeline_state.cpp [hunk 1]
```diff
@@ -347,6 +347,20 @@ void AppendDynamicStateFromSubstate(const Substate &substate, std::vector<VkDyna
     }
 }
 
+std::vector<std::shared_ptr<const PIPELINE_LAYOUT_STATE>> PIPELINE_STATE::PipelineLayoutStateUnion() const {
```
Comment (ncesario-lunarg) [inline layers/pipeline_state.cpp:350]: I originally had this as a span of pointers, returning either the merged layout or a span of the pre-raster and FS layouts, but I think we have to copy the `std::shared_ptr`s to do this safely.

**Review by Tony-LunarG — COMMENTED** [review]
LGTM - the error message is a little hard to parse, but I have no better suggestions.

**Comment by ncesario-lunarg** [comment]
> the error message is a little hard to parse

For sure. I'm open to suggestions. The intent is to convey "this layout is a union of these layouts." Maybe I should spell that out.

**Comment by Tony-LunarG** [comment]
> 

Yes, I think that would help

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM
